### PR TITLE
Add support for use as pre-commit hook

### PR DESCRIPTION
**Pull request by proinsias**
* Adds configuration files needed for `image_optim` to be used as [pre-commit.com](https://pre-commit.com) hook.
* Also adds GitHub Action to regularly run [`pre-commit autoupdate`](https://pre-commit.com/#pre-commit-autoupdate) and create a pull request to update hook versions if needed.

### CODE REVIEW

File: .github/workflows/actions.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+---
+name: GitHub Actions
+
+# yamllint disable-line rule:truthy
+on: [push, pull_request]
```
Comment (toy) [inline .github/workflows/actions.yml:5]: Are those good triggers?

File: .github/workflows/actions.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+---
+name: GitHub Actions
+
+# yamllint disable-line rule:truthy
```
Comment (toy) [inline .github/workflows/actions.yml:4]: As there is no yamllint configured to run, I think this comment shouldn't be here

File: .github/workflows/actions.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+      - name: Cache gems
+        uses: actions/cache@v2.1.4
+        with:
+          path: vendor/bundle
+          key: ${{ runner.os }}-gem-${{ hashFiles('**/Gemfile.lock') }}
+          restore-keys: ${{ runner.os }}-gem-
+
+      - name: Pre-commit autoupdate
+        uses: browniebroke/pre-commit-autoupdate-action@v1.0.0
```
Comment (toy) [inline .github/workflows/actions.yml:24]: There seem to be a suggestion to create a PR after running this

**Review by toy — CHANGES_REQUESTED** [review]
Thanks for contribution.
Unfortunately looks like the action [failed](https://github.com/proinsias/image_optim/actions/runs/733127746).
Could you also please describe what is the action supposed to do? Check if version is up to date with what pre-commit knows as the latest version?

**Comment by proinsias** [comment]
@toy – thanks for the review. I believe I've addressed your comments. I realized I didn't need the GitHub Action as you're not using `pre-commit` for this repo itself.

**Review by toy — APPROVED** [review]

**Comment by toy** [comment]
Thanks for finishing the PR :)
Travis is failing only because failing check on up to date rubies.
Seemed to work when installing from your fork and specifying commit sha for rev, do I understand right that it would be better to create a release (tag) for it to work nicely?

**Comment by toy** [comment]
Created v0.29.0 tag

**Pull request merged**